test/formal/checked_int_add.py

```diff
@@ -0,0 +1,39 @@
+from rule import Rule
+from opcodes import *
+from util import *
+
+"""
+Overflow checked signed integer addition.
+"""
+
+n_bits = 256
+type_bits = 8
+
+while type_bits <= n_bits:
+
+	rule = Rule()
+
+	# Input vars
+	X_short = BitVec('X', type_bits)
+	Y_short = BitVec('Y', type_bits)
+
+	# Z3's overflow and underflow conditions
+	actual_overflow = (BVAddNoOverflow(X_short, Y_short, True) == False)
+	actual_underflow = (BVAddNoUnderflow(X_short, Y_short) == False)
+
+	# cast to full n_bits values
+	X = BVSignedUpCast(X_short, n_bits)
+	Y = BVSignedUpCast(Y_short, n_bits)
+
+	# Constants
+	maxValue = BVSignedMax(type_bits, n_bits)
+	minValue = BVSignedMin(type_bits, n_bits)
+
+	# Overflow and underflow checks in YulUtilFunction::overflowCheckedIntAddFunction
+	overflow_check = AND(ISZERO(SLT(X, 0)), SGT(Y, SUB(maxValue, X)))
+	underflow_check = AND(SLT(X, 0), SLT(Y, SUB(minValue, X)))
+
+	rule.check(actual_overflow, overflow_check != 0)
```

Comment:
`overflow_check` is a Boolean

Reply:
Nope, it isn't :-).

Reply:
Which is good, because it wouldn't be in yul or the evm either - it's a bitwise ``and`` and ``ISZERO`` and ``SGT`` are ``0`` or ``1`` - actually I wonder whether it'll stay like that for typed yul, but I guess it will, since EVM works like that...

Reply:
ugh, true, sorry :p